tests/integration/conformance/conformance_test.go

```diff
@@ -51,6 +51,7 @@ func TestConformance(t *testing.T) {
 			ctx.Fatalf("error loading test cases: %v", err)
 		}
 
+		// TODO how to get the input
 		gal := galley.NewOrFail(ctx, ctx, galley.Config{})
```

Comment:
I think the core framework changes are ok.

However plumbing this config in is a gnarly problem. I can think of a couple of ways to make forward progress here:
- Galley should be getting a sidecar with the new installer. When there is a sidecar, the Galley service should be getting exposed as insecure for localhost. So, switching to using the new installer for your deployments might help the problem here.
- Change your deployment to use insecure Galley. I am not sure if this is feasible for you or not, it might not be, given this effort. Still, throwing out here as a possibility.
- Collect these settings from deployment. When nodeploy is *not* set, then it should be possible to infer whether Galley is insecure or not from istio deployment code/helm values in the Istio test component. When nodeploy is set, then we can have a flag for passing in such settings of the already-deployed-Istio. Then, based on this, we might be able to pick up the secrets/certs from the cluster and store locally (not sure about the details for this latter part, so some experimentation might be needed).

Reply:
Thanks for your response !

- Do you mean this installer: https://github.com/istio/installer or this one: https://github.com/istio/istio/tree/master/install/kubernetes/helm/istio/charts/galley or a completely different one ? I will take a look at it again.
- This is not possible. We wanted to leave the current deployment as it is and just use the new/forming istio conformance tests to validate our istio setup (in a perfect World we would run these test based on a schedule and after all changes at the istio setup (e.g. version upgrades)
-My initial idea was to pass the cert/ca/key per test flags (if they are set we assume Galley want's to talk mTLS and we make use of the files otherwise we use the insecure communication. What you are describing would be step 2 from my perspective to automatically detect if galley is deployed with mTLS and fetch the according certs from K8s (which is pretty is easy).

Reply:
The installer in the first link https://github.com/istio/installer .

Reply:
Thanks, I will take a look if the sidecar handle everything the tests should work without my changes.

Reply:
@ozevren any reason the according patch is missing in the main repo here https://github.com/istio/istio/tree/master/install/kubernetes/helm/istio/charts/galley/templates ? If there is no reason I would like to backport it into this repo since we are using there charts from this repo and it seems like this solves our problems. I will add a note to the conformance tests.